### DISCUSSION TITLE
Semantic predicates and actions in parse tree preview

### DISCUSSION THREAD

**Discussion by clementterrien** [mike-lischke/vscode-antlr4#194] [Ideas]
Hi, 

I've noticed that semantic predicates and actions are not taken into account when we generate a parse tree preview which is very frustrating for debugging, developing or enhancing complex grammars. 

Is there another way to generate a visual rendering of a parser tree which includes semantic predicates and actions ? Maybe after the parse tree listener ? The best would be of course if semantic predicates would appear in the preview tree.

Thanks a lot for your answer, 

Clément Terrien

**Comment by mike-lischke** [comment]
They are shown in the railroad diagram, but not in the parse tree, right.
